_posts/2017-12-07-mixer-spof-myth.md

```diff
@@ -0,0 +1,108 @@
+
+## Mixer: Latency slasher
+
+As we explained above, the Istio sidecars generally have fairly effective first-level caching. They can serve the majority of their traffic from cache. Mixer provides a much greater shared pool of second-level cache, which helps Mixer contribute to a lower average per-request latency.
+
+While it’s busy cutting down latency, Mixer is also inherently cutting down the number of calls your mesh makes to infrastructure backends. Depending on how you’re paying for these backends, this might end up saving you some cash by cutting down the effective QPS to the backends.
+
+## Work ahead
+
+We have opportunities ahead to continue improving the system in many ways.
+
+### Config canaries
+
+Mixer is highly scaled so it is generally resistant to individual instance failures. However, Mixer is still susceptible to cascading failures in the case when a poison configuration is deployed which causes all Mixer instances to crash basically at the same time (yeah, that would be a bad day). To prevent this from happening, config changes can be canaried to a small set of Mixer instances, and then more broadly rolled out. 
+
+Mixer doesn’t yet do canarying of config changes, but we expect this to come online as part of Istio’s ongoing work on reliable config distribution.
+
+### Cache tuning
+
+We have yet to fine-tune the sizes of the sidecar and Mixer caches. This is work will be focused on achieving the highest performance possible using the least amount of resources.
+
+### Cache sharing
+
+At the moment, each Mixer instance operates independently of all other instances. A request handled by one Mixer instance will not leverage data cached in a different instance. We will eventually experiment with a distributed cache such as memcached or Redis in order to provide a much larger mesh-wide shared cache, and further reduce the number of calls to infrastructure backends.
+
+### Sharding
+
+In very large meshes, the load on Mixer can be great. There can be a large number of Mixer instances, each straining to keep caches primed to satisfy incoming traffic. We expect to eventually introduce intelligent sharding such that Mixer instances become slightly specialized in handling particular data streams in order to increase the likelihood of cache hits. We don’t do this yet, but expect to see this down the line.
+
+## Conclusion
+
+Practical experience at Google showed that the model of a slim sidecar proxy and a large caching control plane intermediary hits a sweet spot, delivering excellent perceived availability and latency. We’ve taken the lessons learned there and applied them to create more sophisticated and effective caching, prefetching, and buffering strategies in Istio. We’ve also optimized the communication protocols to reduce overhead when a cache miss does occur.
```

Comment:
In the first line of this para:

and a large **shared** caching control plane ?

Reply:
Done